torch_geometric/explainability/algo/captumexplainer.py

```diff
@@ -0,0 +1,147 @@
+import torch
+
+from torch_geometric.explainability.algo.base import ExplainerAlgorithm
+from torch_geometric.explainability.explanations import Explanation
+from torch_geometric.explainability.utils import CaptumModel
+
+
+class CaptumExplainer(ExplainerAlgorithm):
+    """Wrapper for captum explainers.
+
+    Wraper for captum explainers.
+
+    Note:
+        This algorithm expect the model to be callable with the following
+        signature: model(x, edge_index, **kwargs).
+
+
+    Args:
+        method (str, optional): name of the captum explainer.
+        Defaults to "Saliency".
+    """
+    def __init__(self, method: str = "Saliency") -> None:
+
+        super().__init__()
+        self.method = method
+
+    def explain(
+        self,
+        x: torch.Tensor,
+        edge_index: torch.Tensor,
+        target: torch.Tensor,
+        model: CaptumModel,
+        mask_type: str = "node",
+        **kwargs,
+    ) -> Explanation:
+
+        from captum import attr  # noqa
+
+        # non optimal for now, but captum requires the model to instantiate
+        # the explanation algorithm
+        explainer = getattr(attr, self.method)(model)
+        input, additional_forward_args = self._arange_inputs(
+            x, edge_index, mask_type, **kwargs)
+
+        attributions = explainer.attribute(
+            **self._arange_args(input, target, additional_forward_args))
+
+        return self._create_explanation_from_masks(x, edge_index, attributions)
+
+    def supports(
+        self,
+        explanation_type: str,
+        mask_type: str,
+    ) -> bool:
+        return mask_type != "layers"
+
+    def _arange_inputs(self, x, edge_index, mask_type, **kwargs):
+        """Arrange inputs for captum explainer.
+
+        Create the main input and the additional forward arguments for the
+        captum explainer based on the required type of explanations.
+
+        Args:
+            x (torch.Tensor): node features.
+            edge_index (torch.Tensor): edge indices.
+            mask_type (str): type of explanation to be computed.
+
+        Returns:
+            Tuple[torch.Tensor,Tuple[torch.Tensor]]: main input and additional
+                forward arguments.
+        """
+        input_mask = torch.ones((1, edge_index.shape[1]), dtype=torch.float,
+                                requires_grad=True)
+
+        if mask_type == "node":
+            input = x.clone().unsqueeze(0)
+            additional_forward_args = (edge_index, )
```

Comment:
is x not required for explaining node features?

Reply:
I was thinking that one could try to get node importance in a graph with only edge features (the simplest way would be to first compute the edge attribution and then dispatch it to the nodes it connect)